LeaderboardBackend/Controllers/UsersController.cs

```diff
@@ -47,4 +48,47 @@ public async Task<ActionResult<UserViewModel>> Me() => (await userService.GetUse
             badCredentials => Unauthorized(),
             userNotFound => NotFound()
         );
+
+    [HttpPatch("users/{id}")]
+    [SwaggerOperation(
+        "Updates a user. This request is restricted to administrators, and currently " +
+        "only for banning/unbanning users. " +
+        "This operation is atomic; if an error occurs, the user will not be updated. ",
+        OperationId = "updateUser"
+    )]
+    [SwaggerResponse(204)]
+    [SwaggerResponse(401)]
+    [SwaggerResponse(
+        403,
+        "The requesting User is unauthorized to update users, or this request is an " +
+        "attempt to ban an administrator.",
+        typeof(ProblemDetails)
+    )]
+    [SwaggerResponse(404, Type = typeof(ProblemDetails))]
+    [SwaggerResponse(422, Type = typeof(ValidationProblemDetails))]
+    public async Task<ActionResult> UpdateUser(
+        [FromRoute] Guid id,
+        [FromBody, SwaggerRequestBody(Required = true)] UpdateUserRequest request
+    )
+    {
+        GetUserResult res = await userService.GetUserFromClaims(HttpContext.User);
+
+        if (!res.IsT0)
+        {
+            return Unauthorized();
+        }
+
+        if (res.AsT0.Role is not UserRole.Administrator)
+        {
+            return Forbid();
+        }
```

Comment:
Use attribute authorization for this instead. Also, add an attribute that says this can return 403.

Reply:
I didn't want to use it at first because the attribute uses our old authz scheme. We'll have to fix that eventually

Reply:
Fixed it. The problem now though is that we won't return a reason in the title for this case, breaking the pattern in the other 403 cases. Not sure if users would actually care though.